File: exercise/java/day08/src/main/java/service/ToyProductionService.java
package service;

import domain.Toy;
import domain.ToyRepository;

public class ToyProductionService {
    private final ToyRepository repository;

    public ToyProductionService(ToyRepository repository) {
        this.repository = repository;
    }

    public void assignToyToElf(String toyName) {
        Toy toy = repository.findByName(toyName);
        if (toy != null && toy.getState() == Toy.State.UNASSIGNED) {
            toy.setState(Toy.State.IN_PRODUCTION);
            repository.save(toy);
        }
    }
}

File: exercise/kotlin/day08/src/main/kotlin/service/ToyProductionService.kt
package service

import domain.Toy
import domain.ToyRepository

class ToyProductionService(private val repository: ToyRepository) {
    fun assignToyToElf(toyName: String) {
        val toy = repository.findByName(toyName)
        if (toy != null && toy.state == Toy.State.UNASSIGNED) {
            toy.state = Toy.State.IN_PRODUCTION
            repository.save(toy)
        }
    }
}
